File: 1. Fundamental-Level/CSharp/03. OOP-November-2015/[EXAMS] OOP/OOP-Exam-24-Oct-2014/02. Farmers-Creed/Farmers-Creed-Skeleton/Farmers-Creed/Units/Animals/Animal.cs
namespace FarmersCreed.Units.Animals
{
    using System;
    using FarmersCreed.Interfaces;

    public abstract class Animal : FarmUnit
    {
        protected Animal(string id, int health, int productionQuantity)
            : base(id, health, productionQuantity)
        {
        }

        public virtual void Starve()
        {
            this.Health -= 1;
        }

        public abstract void Eat(IEdible food, int quantity);

        public override string ToString()
        {
            return string.Format("--{0} {1}{2}", this.GetType().Name, this.Id,
                 this.IsAlive ? ", Health: " + this.Health : ", DEAD");
        }
    }
}

File: 1. Fundamental-Level/Java/OOP-Bacis-June-2016/[EX]/03. Static-Members/src/problem5_AnimalClinic/Animal.java
package problem5_AnimalClinic;

public class Animal {
    private String name;
    private String breed;

    public Animal(String name, String breed) {
        this.name = name;
        this.breed = breed;
    }

    public String getName() {
        return this.name;
    }

    public String getBreed() {
        return this.breed;
    }

    @Override
    public String toString() {
        return String.format("%s %s", this.name, this.breed);
    }
}